q2_diversity/_ordination.py

```diff
@@ -28,3 +31,55 @@ def pcoa(distance_matrix: skbio.DistanceMatrix,
 def pcoa_biplot(pcoa: skbio.OrdinationResults,
                 features: pd.DataFrame) -> skbio.OrdinationResults:
     return skbio.stats.ordination.pcoa_biplot(pcoa, features)
+
+
+def tsne(distance_matrix: skbio.DistanceMatrix,
+         number_of_dimensions: int = 3) -> skbio.OrdinationResults:
+
+    data = distance_matrix.data
+    ids = distance_matrix.ids
+
+    if number_of_dimensions == 2:
+        number_of_dimensions = 3
+        tsne = TSNE(2).fit_transform(data)
+        add_zeros = np.zeros((tsne.shape[0], 1), dtype=np.int64)
+        tsneData = np.append(tsne, add_zeros, axis=1)
+
+    else:
+        tsneData = TSNE().fit_transform(data)
+
+    axis_labels = ["TSNE%d" % i for i in range(1, number_of_dimensions + 1)]
+    eigenvalues = [0 for i in axis_labels]
+
+    return skbio.OrdinationResults(
+        short_method_name="T-SNE",
+        long_method_name="t-distributed stochastic neighbor embedding",
+        eigvals=pd.Series(eigenvalues, index=axis_labels),
+        samples=pd.DataFrame(tsneData, index=ids, columns=axis_labels),
```

Comment:
Emperor expect a non empty proportion_explained attribute in OrdinationResults. Is this computed via tsne? If not, `proportion_explained=pd.Series(None, index=axis_labels)` might do the trick here

Reply:
Thank you very much :)